backend/app/testbot/server.py

```
@@ -3,16 +3,124 @@
+    # Extract the substring that should form the JSON object
+    json_string = input_string[idx_start: idx_end + 1]
+    
+    try:
+        data = json.loads(json_string)
+    except json.JSONDecodeError as e:
+        raise ValueError("The extracted string is not valid JSON.") from e
+    
+    return clean_json_response(data)
+    
+
+async def process_query(client_id: str, chatbot_id: str, query: str) -> Dict[str, Any]:
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)